src/sound/YMF278.cc

```diff
@@ -908,8 +909,10 @@ void YMF278::reset(EmuTime::param time)
 // for the region 0x000000-0x1FFFFF. (But this routine does not handle ROM).
 unsigned YMF278::getRamAddress(unsigned addr) const
 {
-	addr -= 0x200000; // RAM starts at 0x200000
-	if (regs[2] & 2) [[unlikely]] {
+	if (romEnabled) {
+		addr -= 0x200000; // RAM starts at 0x200000
+	}
```

Comment:
My _guess_ would be that if you enable/disable the sample-ROM, this will only (un)hide the RAM that's present in the first 2MB. And the RAM present from 2MB to 4MB remains unchanged.

Though this implementation will instead shift the lower part of the RAM to be visible in the upper part of the address space (and make the upper part of the RAM inaccessible).